File: Problems/No.56_Merge_Intervals/merge_intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        
        // parameters
        std::vector<std::vector<int>> merged_intervals;

        // start computing
        std::sort(intervals.begin(), intervals.end());
        for (auto interval: intervals) {

            if (merged_intervals.empty() || merged_intervals.back()[1] < interval[0])
                merged_intervals.push_back(interval);
            else merged_intervals.back()[1] = std::max(merged_intervals.back()[1], interval[1]);

        }

        return merged_intervals;

    }
};

File: Problems/No.56_Merge_Intervals/merge_intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """

        # parameters
        merged_intervals = []

        # start computing
        intervals.sort()
        for interval in intervals:
            if (not merged_intervals or merged_intervals[-1][1] < interval[0]):
                merged_intervals.append(interval)
            else:
                merged_intervals[-1][1] = max(merged_intervals[-1][1], interval[1])
        
        return merged_intervals
